include/flatbuffers/flatbuffers.h

```diff
@@ -434,7 +434,14 @@ template<typename T, uint16_t length> class Array {
 
   // Change elements if you have a non-const pointer to this object.
   void Mutate(uoffset_t i, const T &val) {
+    #if defined(_MSC_VER)
+	  __pragma(warning(push))
+	  __pragma(warning(disable:4127)) // C4127: conditional expression is constant
```

Comment:
You can use on-line suppression without push/pop:
```cpp
// base.h
// Suppress false positive warning C4127: expression is constant`.
// This false warning generated under MSVC201/MSVC2013 only.
// The suppress	pushes the current state of the pragma on the stack,
// disables the specified warning for the next line,
// and then pops the warning stack so that the pragma state is reset.
#if defined _MSC_VER && _MSC_VER < 1900
  #define __suppress_msvc_C4127__() __pragma(warning(suppress:4127))
#else
  #define __suppress_msvc_C4127__()
#endif


__suppress_msvc_C4127__()
if (flatbuffers::is_scalar<T>::value) {
...
} else { ... }
```

Reply:
After some digging I found `IsConstTrue` for that purpose :)